lib/engine/round/operating.rb

```diff
@@ -364,10 +393,14 @@ def change_share_price(direction)
       end
 
       def buy_train(entity, train, price, exchange)
+        # Check if the train is actually buyable in the current situation
+        raise GameError, 'Not a buyable train' unless buyable_trains.include? train
+
         remaining = price - entity.cash
 
-        if train.from_depot? && remaining.positive? && train.name == @depot.min_depot_train.name && must_buy_train?
+        if remaining.positive?
```

Comment:
there's one more bug i think that i'd like to make sure we solve.

i believe right now, if you are buying a train that causes a phase shift that rusts all of your trains, the game things you're doing an e-buy and allows you to contribute funds to buy the next train. can we prevent that from happening?

Reply:
ahh, yeah. the phase rusting is applied before the buying of trains

Reply:
that breaks 3 of the rspec games, will need a pinning for certain

Reply:
i'll hunt for some new rspec games. hopefully there'll be some that haven't exploited it.

Reply:
Nope, the implementation needs more work. It needs to allow it if you didn't have a train.

Reply:
All the existing rspec games still pass, and I added another chesapeake one.